feature-libs/product/events/src/product-event.builder.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { TestBed } from '@angular/core/testing';
+import { Action, ActionsSubject } from '@ngrx/store';
+import {
+  createFrom,
+  EventService,
+  Product,
+  ProductSearchPage,
+  ProductSearchService,
+  ProductService,
+} from '@spartacus/core';
+import { PageVisitedEvent } from '@spartacus/storefront';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { ProductEventBuilder } from './product-event.builder';
+import {
+  CategoryPageResultsEvent,
+  ProductDetailsPageEvent,
+  SearchPageResultsEvent,
+} from './product.events';
+
+const productGetBehavior = new BehaviorSubject<Product>(undefined);
+class MockProductService {
+  get = () => productGetBehavior;
+}
+
+const getResultsBehavior = new BehaviorSubject<ProductSearchPage>(undefined);
+class MockProductSearchService {
+  getResults = () => getResultsBehavior;
+}
+
+interface ActionWithPayload extends Action {
+  payload: any;
+}
+
+describe('ProductEventModule', () => {
+  let eventService: EventService;
+  let actions$: Subject<ActionWithPayload>;
+
+  beforeEach(() => {
+    actions$ = new Subject();
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: ActionsSubject, useValue: actions$ },
+        { provide: ProductService, useClass: MockProductService },
+        { provide: ProductSearchService, useClass: MockProductSearchService },
+      ],
+    });
+
+    TestBed.inject(ProductEventBuilder); // register events
+    eventService = TestBed.inject(EventService);
+  });
+
+  it('SearchPageResultsEvent', () => {
```

Comment:
Would be nice to also test not obvious behavior, so:
- event will be emitted also if only search will happen
- event won't be emitted if navigated out (even if search will happen)

Reply:
I don't understand the second point?